### ISSUE TITLE
[Joins] Error on indexing reference field with enabled nested fields

### ISSUE THREAD

**Issue by mitallast** [typesense/typesense#1664]
## Description

Error while indexing document with reference field as nested.

## Steps to reproduce

1) Create origin collection

```shell
curl -sv -X POST --location "http://localhost:8108/collections" \
    -H "x-typesense-api-key: testapikey" \
    -H "Content-Type: application/json" \
    -d '{
          "name": "foo",
          "fields": [
            {"name": "title", "type": "string"}
          ]
        }'
        
```

```text
< HTTP/1.1 201 Created
< Connection: keep-alive
< access-control-allow-origin: *
< content-type: application/json; charset=utf-8
< transfer-encoding: chunked
< 
* Connection #0 to host localhost left intact
{"created_at":1712826694,"default_sorting_field":"","enable_nested_fields":false,"fields":[{"facet":false,"index":true,"infix":false,"locale":"","name":"title","optional":false,"sort":false,"stem":false,"type":"string"}],"name":"foo","num_documents":0,"symbols_to_index":[],"token_separators":[]}
```

2) Create collection with reference field as nested

```shell
curl -sv -X POST --location "http://localhost:8108/collections" \
    -H "x-typesense-api-key: testapikey" \
    -H "Content-Type: application/json" \
    -d '{
          "name": "bar",
          "fields": [
            {
              "name": "foo.foo_id",
              "type": "string",
              "reference": "foo.id"
            },
            {
              "name": "title",
              "type": "string"
            }
          ],
          "enable_nested_fields": true
        }'
```

```text
< HTTP/1.1 201 Created
< Connection: keep-alive
< access-control-allow-origin: *
< content-type: application/json; charset=utf-8
< transfer-encoding: chunked
< 
* Connection #0 to host localhost left intact
{"created_at":1712826804,"default_sorting_field":"","enable_nested_fields":true,"fields":[{"facet":false,"index":true,"infix":false,"locale":"","name":"foo.foo_id","optional":false,"reference":"foo.id","sort":false,"stem":false,"type":"string"},{"facet":false,"index":true,"infix":false,"locale":"","name":"title","optional":false,"sort":false,"stem":false,"type":"string"}],"name":"bar","num_documents":0,"symbols_to_index":[],"token_separators":[]}
```

3) Add origin document

```shell
curl -sv -X POST --location "http://localhost:8108/collections/foo/documents" \
    -H "x-Typesense-Api-Key: testapikey" \
    -H "Content-Type: application/json" \
    -d '{
          "id": "123",
          "title": "hello"
        }'
```

```text
< HTTP/1.1 201 Created
< Connection: keep-alive
< access-control-allow-origin: *
< content-type: application/json; charset=utf-8
< transfer-encoding: chunked
< 
* Connection #0 to host localhost left intact
{"id":"123","title":"hello"}
```

4) Try to add document with reference to origin existing document

```shell
curl -sv -X POST --location "http://localhost:8108/collections/bar/documents" \
    -H "x-Typesense-Api-Key: testapikey" \
    -H "Content-Type: application/json" \
    -d '{
          "id": "124",
          "title": "hello",
          "foo": {
            "foo_id": "123"
          }
        }'
```

```text
< HTTP/1.1 404 Not Found
< Connection: keep-alive
< access-control-allow-origin: *
< content-type: application/json; charset=utf-8
< transfer-encoding: chunked
< 
* Connection #0 to host localhost left intact
{"message":"Field `foo.foo_id_sequence_id` not found."}
```

## Expected Behavior

Success on indexing document with reference field as nested

## Actual Behavior

Error with unexpected `_foo.foo_id_sequence_id` field

## Metadata

**Typesense Version**: 26.0, running in docker
**OS**: ubuntu 22.04

**Comment by happy-san** [comment]
This issue is fixed in `27.0.rc9`.

**Comment by kaiden-tradeblock-us** [comment]
> This issue is fixed in `27.0.rc9`.

I've tried with the published Docker images for both [27.0.rc9](https://hub.docker.com/layers/typesense/typesense/27.0.rc9/images/sha256-2674de0d58c212da0b0260758a6c8e91479014f9020f3f96e039ce8212dad228?context=explore) and the latest image (currently [27.0.rc29](https://hub.docker.com/layers/typesense/typesense/27.0.rc29/images/sha256-7f4343b19b43e7229eeb67447f83bad1de9cac3e1e5a16e1a4797475046104bd?context=explore)), but am still running into the same error with indexing a referenced nested field.

Running on a M1.

**Comment by happy-san** [comment]
@kaiden-tradeblock-us You'll have to add `foo` field in the schema as well.
```bash
curl -sv -X POST --location "http://localhost:8108/collections" \
    -H "x-typesense-api-key: testapikey" \
    -H "Content-Type: application/json" \
    -d '{
          "name": "bar",
          "fields": [
            {
              "name": "foo",
              "type": "object"
            },
            {
              "name": "foo.foo_id",
              "type": "string",
              "reference": "foo.id"
            },
            {
              "name": "title",
              "type": "string"
            }
          ],
          "enable_nested_fields": true
        }'
```
